### PR TITLE
Documentation improvements (`strategies/generic`)

### PR DESCRIPTION
**Pull request by Cerebrovinny**

### CODE REVIEW

File: bw2io/strategies/generic.py [hunk 1]
```diff
@@ -74,7 +150,8 @@ def link_iterable_by_fields(
 
 
 def assign_only_product_as_production(db):
-    """Assign only product as reference product.
+    """
+    Assign only product as reference product.
```
Comment (michaelweinold) [inline bw2io/strategies/generic.py:165]: Not immediately clear what this means.

File: bw2io/strategies/generic.py [hunk 2]
```diff
@@ -148,14 +316,81 @@ def tupleize_categories(db):
 
 
 def drop_unlinked(db):
-    """This is the nuclear option - use at your own risk!"""
+    """
+    Remove all exchanges in a given database that don't have inputs.
+
+    This is the nuclear option - use at your own risk!
```
Comment (michaelweinold) [inline bw2io/strategies/generic.py]: Move to 'Notes' and maybe add '⚠️'.

File: bw2io/strategies/generic.py [hunk 3]
```diff
@@ -171,14 +406,57 @@ def normalize_units(db):
...
-    """Add database name to datasets"""
+    """
+    Add a database name to the datasets.
+
+    Parameters
+    ----------
+    db : list[dict]
+        The list of datasets to add the database name to.
+    name : str
+        The name of the database to be added.
```
Comment (michaelweinold) [inline bw2io/strategies/generic.py]: "Name __of the database to be added__" implies that this functions adds a database. As far as I see it only adds the name. Best to reformulate this.

File: bw2io/strategies/generic.py [hunk 4]
```diff
@@ -11,6 +11,36 @@
 
 
 def format_nonunique_key_error(obj, fields, others):
+    """
+    This function takes an object that can't be uniquely linked to the target
```
Comment (michaelweinold) [inline bw2io/strategies/generic.py]: ...same issue as [here](https://github.com/brightway-lca/brightway2-io/pull/175#discussion_r1153681421): You mix imperative ("Perform XYZ") and declarative ("This function performs XYZ") throughout.

**Comment by Cerebrovinny** [comment]
@michaelweinold I did many changes before you do the review so please proceed the review again for this module

File: bw2io/strategies/generic.py (hunk 4 above)
Comment (Cerebrovinny) [reply bw2io/strategies/generic.py]: this is resolved on the last changes, I made the PR and forgot to push some local changes after

**Pull request merged**